llvm/include/llvm/Analysis/WithCache.h

```diff
@@ -0,0 +1,97 @@
+//===- llvm/Analysis/WithCache.h - KnownBits cache for pointers -*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Store a pointer to any type along with the KnownBits information for it
+// that is computed lazily (if required).
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_ANALYSIS_WITHCACHE_H
+#define LLVM_ANALYSIS_WITHCACHE_H
+
+#include "llvm/IR/Value.h"
+#include "llvm/Support/KnownBits.h"
+#include <type_traits>
+
+namespace llvm {
+struct SimplifyQuery;
+}
+
+llvm::KnownBits computeKnownBits(const llvm::Value *V, unsigned Depth,
+                                 const llvm::SimplifyQuery &Q);
```

Comment:
This function needs to be part of the `llvm` namespace as well.

Reply:
What's the error message? You also need to adjust the corresponding definitions in ValueTracking.cpp.

Reply:
Ah yeah, that was it. I had originally moved the function into the namespace but then I reverted it, because moving it causes a whole lot of errors when `::computeKnownBits` is used to call it.

Reply:
Yeah, need to drop the `::` prefix from those calls as well.

Reply:
Okay, done.